src/main/java/org/passay/rule/IllegalRegexRule.java

```diff
@@ -100,20 +101,25 @@ public Pattern getPattern()
   public RuleResult validate(final PasswordData passwordData)
   {
     PassayUtils.assertNotNullArg(passwordData, "Password data cannot be null");
-    final List<RuleResultDetail> details = new ArrayList<>();
-    final Matcher m = pattern.matcher(passwordData.getPassword());
-    final Set<String> matches = new HashSet<>();
-    while (m.find()) {
-      final String match = m.group();
-      if (!matches.contains(match)) {
-        details.add(new RuleResultDetail(ERROR_CODE, createRuleResultDetailParameters(match)));
-        if (!reportAllFailures) {
-          break;
+    final CharBuffer password = passwordData.getPassword().toCharBuffer();
+    try {
+      final List<RuleResultDetail> details = new ArrayList<>();
+      final Matcher m = pattern.matcher(password);
```

Comment:
If `UnicodeString` implemented `CharSequence`, you could use it here without a conversion.